app/file-upload/server/lib/FileUpload.js

```diff
@@ -389,6 +392,29 @@ export const FileUpload = {
 
 		request.get(fileUrl, (fileRes) => fileRes.pipe(res));
 	},
+
+	addJWTToFileUrlIfNecessary(message) {
+		if (!settings.get('FileUpload_ProtectFiles') || !settings.get('FileUpload_Enable_json_web_token_for_files') || !message.file) {
+			return message;
+		}
+		const jwt = generateJsonWebToken({
+			rid: message.rid,
+			userId: message.u._id,
+			fileId: message.file._id,
+		}, settings.get('FileUpload_json_web_token_secret_for_files'));
+
+		if (message.attachments && Array.isArray(message.attachments) && message.attachments.length) {
```

Comment:
...and this code looks to be a good `helper method` which would return the `message` object.